src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSourceConfig.java

```diff
@@ -0,0 +1,238 @@
+                                                 "Please specify a project id.", PROJECT);
+    }
+    if (!containsMacro(INSTANCE) && Strings.isNullOrEmpty(instance)) {
+      throw new InvalidConfigPropertyException("Instance ID must be specified", INSTANCE);
+    }
+    String serviceAccountFilePath = getServiceAccountFilePath();
+    if (!containsMacro(SERVICE_ACCOUNT_FILE_PATH) && null != serviceAccountFilePath) {
+      File serviceAccountFile = new File(serviceAccountFilePath);
+      if (!serviceAccountFile.exists()) {
+        throw new InvalidConfigPropertyException(
+          String.format("Service account file '%s' does not exist", serviceAccountFilePath),
+          SERVICE_ACCOUNT_FILE_PATH
+        );
+      }
+    }
+    if (!containsMacro(ON_ERROR) && null != onError) {
+      if (Strings.isNullOrEmpty(onError)) {
+        throw new InvalidConfigPropertyException("Error handling must be specified", ON_ERROR);
+      }
+      if (null == ErrorHandling.fromDisplayName(onError)) {
+        throw new InvalidConfigPropertyException("Invalid record error handling strategy name", ON_ERROR);
+      }
+    }
+    if (!containsMacro(SCHEMA)) {
+      Schema parsedSchema = getSchema();
+      if (null == parsedSchema) {
+        throw new InvalidConfigPropertyException("Schema must be specified", SCHEMA);
+      }
+      if (Schema.Type.RECORD != parsedSchema.getType()) {
+        String message = String.format("Schema is of invalid type '%s'. The schema must be a record.",
+                                       parsedSchema.getType());
+        throw new InvalidConfigPropertyException(message, SCHEMA);
+      }
+      List<Schema.Field> fields = parsedSchema.getFields();
+      if (null == fields || fields.isEmpty()) {
+        throw new InvalidConfigPropertyException("Schema should contain fields to map", SCHEMA);
+      }
+      for (Schema.Field field : fields) {
+        Schema.Type fieldType = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+        if (!SUPPORTED_FIELD_TYPES.contains(fieldType)) {
+          String supportedTypes = SUPPORTED_FIELD_TYPES.stream()
+            .map(Enum::name)
+            .map(String::toLowerCase)
+            .collect(Collectors.joining(", "));
+          String errorMessage = String.format("Wrong type '%s' for field '%s'. Supported types are: '%s'",
+                                              fieldType, field.getName(), supportedTypes);
+          throw new InvalidConfigPropertyException(errorMessage, SCHEMA);
+        }
+        if (field.getName().equals(keyAlias)) {
+          // key should be defined in schema but will not be retrieved from column families.
+          continue;
+        }
+        if (!HBaseColumn.isValidName(field.getName())) {
+          String errorMessage = String.format("Wrong name for field '%s'. Required format is '<family>__<qualifier>'",
```

Comment:
this is a little dirty because both this config class and HBaseColumn now need to know what the expected format is. It's cleaner to have HBaseColumn throw an exception with this message and to have this class catch the exception and pass the message to InvalidConfigPropertyException.